rotkehlchen/externalapis/hyperliquid.py

```diff
@@ -0,0 +1,122 @@
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.fval import FVal
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval
+from rotkehlchen.types import ChecksumEvmAddress, Location
+from rotkehlchen.utils.network import create_session
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def asset_from_hyperliquid(symbol: str) -> Asset:
+    """Fetch asset from hperliquid location
+
+    May raise:
+    - UnsupportedAsset
+    - UnknownAsset
+    """
+    return symbol_to_asset_or_token(GlobalDBHandler.get_assetid_from_exchange_name(
+        exchange=Location.HYPERLIQUID,
+        symbol=symbol,
+        default=symbol,
+    ))
+
+
+class Hyperliquid:
+
+    def __init__(self) -> None:
+        self.base_url = 'https://api.hyperliquid.xyz'
+        self.session = create_session()
+        self.arb_usdc = Asset('eip155:42161/erc20:0xaf88d065e77c8cC2239327C5EDb3A432268e5831')
+
+    def _query(
+            self,
+            address: ChecksumEvmAddress,
+            account_type: Literal['clearinghouseState', 'spotClearinghouseState'],
+    ) -> dict[str, Any]:
+        """Query hyperliquid for balances.
+
+        May raise:
+            - RemoteError
+        """
+        log.debug(f'Querying hyperliquid balances at {self.base_url}/info with {account_type=} and {address=}')  # noqa: E501
+        try:
+            response = self.session.post(
+                url=f'{self.base_url}/info',
+                json={'type': account_type, 'user': address},
+            )
+            data = response.json()
+        except requests.exceptions.RequestException as e:
+            log.error(
+                f'Query to hyperliquid for {address} and {account_type=} failed due to '
+                f'{e}. Skipping',
+            )
+            raise RemoteError(
+                f'Failed to query balances of {account_type} for {address} due to {e}',
+            ) from e
+
+        if response.status_code != HTTPStatus.OK:
+            log.error(
+                f'Failed to query {account_type} balances for {address} with status '
+                f'code {response.status_code} and text {response.text}',
+            )
+            raise RemoteError(f'Failed to query balances of {account_type} for {address} with status code {response.status_code}')  # noqa: E501
+
+        return data
+
+    def query_balances(self, address: ChecksumEvmAddress) -> dict[Asset, FVal]:
+        """
+        Queries both spot and perp balances since they are returned in two different endpoints.
+        Hyperliquid has like two `accounts` for each address, one for spot and one for perpetuals.
```

Comment:
maybe add a reference to the docs.
https://hyperliquid.gitbook.io/hyperliquid-docs/for-developers/api/info-endpoint/spot#retrieve-a-users-token-balances

https://hyperliquid.gitbook.io/hyperliquid-docs/for-developers/api/info-endpoint/perpetuals#retrieve-users-perpetuals-account-summary